components/TemplateGrid/TemplateGridWrapper.vue

```diff
@@ -114,20 +141,40 @@ export default {
                 }, { hiddenCategories: [], visibleCategories: [] });
 
                 this.setHiddenItem({ key: id, value: hiddenCategories });
-                this.setTreeWhenCollapse({ tree: visibleCategories, index: rowValue });
+                this.setGridWhenCollapse({ data: visibleCategories, index: rowValue });
                 this.setExpandItem({ index: rowValue, value: true });
             } else {
-                this.setTreeWhenExpand({ id, index: rowValue });
+                this.setGridWhenExpand({ id, index: rowValue });
                 this.removeHiddenItem(id);
                 this.setExpandItem({ index: rowValue, value: false });
             }
         },
+        connectionLineStyle({ id, row, parent }) {
+            const children = this.filteredGridData.filter(e => e.parent === parent);
+            const connectionHeight = this.rowsHeight * (
+                row - (children.length ? children[0].row : 0) + 1
+            );
+            const borderStyle = id === 'ghost_item' ? 'dashed' : 'solid';
+            const lineCoordinates = {
```

Comment:
if (root) - one condition instead of two doing same think

Reply:
Also why an object for lineCoordinates ? Width has nothing common with coordinate